### DISCUSSION TITLE
Random ideas and future remake

### DISCUSSION THREAD

**Discussion by davidemerli** [davidemerli/human-aoc#4] [Ideas]
Feel free to share anything related to the project that would make it:
- more easily accessible
- more engaging
- look cooler
- include random swag in any way

No idea is 'too much' or unfeasible by default, just throw that shit down here

**Comment by dario-passarello** [comment]
# Advent of Code Golfing

A parallel leaderboard using as the metric the shortest solution.
It uses the classic star system but more points are awareded to the shortest solution instead of the fastest submit

Potential problems and solutions:
* Comparing different languages in the same code golfing competition
  * That's a problem of the users. If they choose java for a code golfing competition, how could they hope to win?

* The solution correctness is hard to verify automatically 
    * Honor system, we already trust the players to visit the page first for starting the timer
    
 See also: https://github.com/Starwort/advent-of-golf-2022

**Comment by dario-passarello** [comment]
# Language Bar



![Screenshot from 2023-11-30 13-24-43](https://github.com/davidemerli/human-aoc/assets/55185166/a4e32845-4711-4ea3-8992-34c8e593c1d1)


What about a bar like this to show the favourite languages used by the player. The bar is calculated by analyzing the uploaded code solutions

**Comment by Hakueiryu** [comment]
### Notification System on Solutions
Implement a mechanism that allows users to know if another user has liked or commented one of their solves (e.g. email, notification bell within the app, ...)
Currently users are required to login and perform a manual check to see any reply

**Comment by Alexdruso** [comment]
# Solutions benchmark

Have a benchmark setup similar to https://github.com/marcelblijleven/adventofcode. Could be as easy as to ask players to use this template, cloning all their repos and running generate-benchmark on each to get the numbers.

Support for other languages might be harder and less standardized.

**Reply by davidemerli** [reply]
how about having a dockerfile for each language (can be maintained by all the ones interested in using a specific language), then having the server execute everything when a new solution is proposed? (plus timeouts, plus execution limits per user))?

**Comment by kientuong114** [comment]
### Sharing code from GitHub

Instead of copying and pasting the code, it would be nice to be able to paste a (public) GitHub link and have it automatically put as a writeup. Most people already have their code pushed to Github, making it easier for them to share.

**Comment by MarcoBalossini** [comment]
## Checkpoint

Insert a "Start" button to start the second star's timer, instead of doing it right after the completion of the first star.

**Comment by davidemerli** [comment]
Add flares to users, like reddit!! @MarcoBalossini will be `the bruteforcer` for example
